src/Text/Pandoc/Readers/Docx.hs

```diff
@@ -660,18 +686,14 @@ bodyPartToBlocks (Tbl cap _ look parts@(r:rs)) = do
       nonEmpty [] = Nothing
       nonEmpty l  = Just l
       rowLength :: Docx.Row -> Int
-      rowLength (Docx.Row c) = length c
+      rowLength (Docx.Row _ c) = length c
 
-  let toRow = Pandoc.Row nullAttr . map simpleCell
-      toHeaderRow l = if null l then [] else [toRow l]
+  let toHeaderRow :: Pandoc.Row -> [Pandoc.Row]
+      toHeaderRow row@(Pandoc.Row _ cs) = if null cs then [] else [row]
 
-  -- pad cells.  New Text.Pandoc.Builder will do that for us,
-  -- so this is for compatibility while we switch over.
-  let cells' = map (\row -> toRow $ take width (row ++ repeat mempty)) cells
+  hdrCells <- fmap (>>= toHeaderRow) (traverse rowToRow hdr)
```

Comment:
@tshort Also, since the header now allows multiple rows, this should be using `rowsToRows` instead of `rowToRow`, which is no longer needed.